rust/xaynet-core/src/mask/masking.rs

```diff
@@ -122,12 +145,22 @@ impl Aggregation {
             return Err(UnmaskingError::NoModel);
         }
 
-        if self.nb_models > self.object.config.model_type.max_nb_models() {
+        if self.nb_models > self.object.vector.config.model_type.max_nb_models() {
             return Err(UnmaskingError::TooManyModels);
         }
 
-        if self.object.config != mask.config || self.object_size != mask.data.len() {
-            return Err(UnmaskingError::MaskMismatch);
+        if self.nb_models > self.object.scalar.config.model_type.max_nb_models() {
+            return Err(UnmaskingError::TooManyScalars);
+        }
+
+        if self.object.vector.config != mask.vector.config
+            || self.object_size != mask.vector.data.len()
+        {
+            return Err(UnmaskingError::MaskManyMismatch);
```

Comment:
The naming is a bit inconsistent between model/scalar vs many/one. One one side we have `TooManyScalars` and `TooManyModels`, and here `MaskManyMismatch` and `MaskOneMismatch`. There are other places where that is the case. Is the motivation for introducing the many/one terminology to be broader than the ML specific term "model"? I like the idea but I think we should have a clear separation of where we use a specific terminology. For instance we could say that in `xaynet-core` there shouldn't be any mention of "model", while in `xaynet-client` and `xaynet-server` which are domain specific, we should only use the model/scalar terminology (it would even be better if we could find a more specific term than "scalar" imo).

On the same topic, I have a slight preference for vector/scalar over many/one. It's not domain specific but fits better the nature of the data we're referring to. Wdyt?

Reply:
yes I agree the naming isn't the best at the moment. OT1H I had introduced new terminology, OTOH I was also trying to not deviate too much from the existing names here for these errors (which are not named the best either!). Let me revise this separately if that's ok and we can continue the discussion there. And yes, the idea of Many vs One was to more abstractly capture the structure of masked models / scalars and model masks / scalar masks. Vector / Scalar would also have worked but scalar clashes with the more domain specific use of the term, as you also noted.